contracts/JBTiered721DelegateStore.sol

```diff
@@ -812,13 +825,13 @@ contract JBTiered721DelegateStore is IJBTiered721DelegateStore {
 
       // Set the reserved token beneficiary if needed.
       if (_tierToAdd.reservedTokenBeneficiary != address(0))
-        if (_tierToAdd.shouldUseReservedTokenBeneficiaryAsDefault)
+        if (_tierToAdd.shouldUseReservedTokenBeneficiaryAsDefault && defaultReservedTokenBeneficiaryOf[msg.sender] != _tierToAdd.reservedTokenBeneficiary)
           defaultReservedTokenBeneficiaryOf[msg.sender] = _tierToAdd.reservedTokenBeneficiary;
         else _reservedTokenBeneficiaryOf[msg.sender][_tierId] = _tierToAdd.reservedTokenBeneficiary;
```

Comment:
This isn't tested @viraj124  and got a doubt:

if `_tierToAdd.shouldUseReservedTokenBeneficiaryAsDefault` is true but `defaultReservedTokenBeneficiaryOf[msg.sender] == _tierToAdd.reservedTokenBeneficiary`
-> defaultReservedBenef is already set, so good, but we end up setting  _reservedTokenBeneficiaryOf[msg.sender][_tierId] too (iic, defautReserved isn't read anymore then, even if we change it later on - Ln#687)

Same bellow for royalty

Reply:
nice catch